src/ui/Help.tsx

```diff
@@ -40,21 +26,171 @@ export function Help(): JSX.Element {
   )
 }
 
-export function Keys(props: HTMLAttributes<HTMLDivElement>): JSX.Element {
+function Button({ value, onClick }: { value: string; onClick: () => void }): JSX.Element {
+  const [isHovered, setIsHovered] = useState(false)
   return (
-    <div {...props}>
-      {controlOptions.map(({ keys, action }) => (
-        <div className="popup-item" key={action}>
-          <div>{action}</div>
-          <div className="popup-item-keys">
-            {keys.map((key) => (
-              <span className="popup-item-key" key={key}>
-                <span>{key}</span>
-              </span>
-            ))}
-          </div>
-        </div>
-      ))}
+    <button
+      onClick={onClick}
+      className={`popup-item-key${isHovered ? ' popup-item-key_hovered' : ''}`}
+      onMouseEnter={() => setIsHovered(true)}
+      onMouseLeave={() => setIsHovered(false)}
+    >
+      <span>{value}</span>
+    </button>
+  )
+}
+
+function Row({
+  action,
+  keys,
+  hasError,
+  onRemove,
+  onAdd,
+}: {
+  action: string
+  keys: Key[]
+  hasError: boolean
+  onAdd: (action: string) => void
+  onRemove: (action: string, name: string) => void
+}): JSX.Element {
+  const [isHovered, setIsHovered] = useState(false)
+  return (
+    <div
+      className={`popup-item${isHovered ? ' hovered' : ''}${hasError ? ' with-error' : ''}`}
+      onMouseOver={() => {
+        setIsHovered(true)
+      }}
+      onMouseLeave={() => {
+        setIsHovered(false)
+      }}
```

Comment:
use `:hover` instead of using state to handle this.
https://developer.mozilla.org/en-US/docs/Web/CSS/:hover

Reply:
with css-only approach it is impossible to manage the presence of the button in DOM (only being able to hide it visually)

Reply:
I'm pretty sure there are a few ways of doing this.
what about using `opacity: 0`?

Reply:
holding this elements in the DOM will also hold them in vDOM what is non-optimal due to the fact that only one line can be hovered in one moment (but yes, it is kind of trade of between holding nodes in memory and holding callbacks end event listeners in it)

Reply:
yes I understand but the css `:hover` is soo much faster and less buggy. Using state will cause an unnecessary rerender.

Reply:
see your point and accepting it